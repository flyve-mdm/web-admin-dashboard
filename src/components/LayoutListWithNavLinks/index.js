import React from 'react'
import { NavLink } from 'react-router-dom'


const ListWithNavLinks = ({routes, rootPath, children}) => {
    return (
        <div className="layout_list_with_navlinks-block">
            <nav>
                <ul>
                    {routes.map((routes, i) => {
                        return (
                            <li key={i}>
                                <NavLink 
                                    exact
                                    to={
                                        typeof(rootPath) === "string" 
                                        ? routes.path === '/'
                                            ? rootPath
                                            : rootPath + routes.path
                                        : routes.path
                                    }
                                    activeClassName='--active'>
                                    {routes.name}
                                </NavLink>
                            </li>
                        )
                    })}
                </ul>
            </nav>
            <article>
                {children}
            </article>
        </div>
    );
}

export default ListWithNavLinks