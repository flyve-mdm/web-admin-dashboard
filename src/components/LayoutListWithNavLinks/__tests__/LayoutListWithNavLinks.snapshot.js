/*
 *   Copyright © 2018 Teclib. All rights reserved.
 *
 *   This file is part of web-mdm-dashboard
 *
 * web-mdm-dashboard is a subproject of Flyve MDM. Flyve MDM is a mobile
 * device management software.
 *
 * Flyve MDM is free software: you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or (at your option) any later version.
 *
 * Flyve MDM is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * ------------------------------------------------------------------------------
 * @author     Gianfranco Manganiello (gmanganiello@teclib.com)
 * @author     Hector Rondon (hrondon@teclib.com)
 * @copyright  Copyright © 2018 Teclib. All rights reserved.
 * @license    GPLv3 https://www.gnu.org/licenses/gpl-3.0.html
 * @link       https://github.com/flyve-mdm/web-mdm-dashboard
 * @link       http://flyve.org/web-mdm-dashboard
 * @link       https://flyve-mdm.com
 * ------------------------------------------------------------------------------
 */

import React from 'react'
import renderer from 'react-test-renderer'
import {
  Router,
} from 'react-router-dom'
import history from '../../../shared/history'

import LayoutListWithNavLinks from '../index'

describe('LayoutListWithNavLinks', () => {
  beforeAll(() => {
    global.Animation = class Animation {
      play = () => {}
    }
    global.KeyframeEffect = class KeyframeEffect {}
  })

  afterAll(() => {
    global.Animation = undefined
    global.KeyframeEffect = undefined
  })

  test('renders LayoutListWithNavLinks', () => {
    const component = renderer.create(
      <Router history={history}>
        <LayoutListWithNavLinks
          rootPath=""
          history={history}
          routes={[
            {
              path: '/',
              name: 'test 1',
              component: <ul />,
              exact: true,
            },
            {
              path: '/test',
              name: 'test 2',
              component: <ol />,
              exact: true,
            },
          ]}
        >
          <div />
        </LayoutListWithNavLinks>
      </Router>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
