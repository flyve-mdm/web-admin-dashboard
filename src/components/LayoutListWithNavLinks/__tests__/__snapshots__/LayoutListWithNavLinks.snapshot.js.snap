// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutListWithNavLinks renders LayoutListWithNavLinks 1`] = `
<div
  className="flex-block flex-block--with-scroll flex-block--with-content-pane"
>
  <nav
    className="flex-block__list navlinks"
    style={Object {}}
  >
    <ul>
      
      <li>
        <a
          aria-current={null}
          href="/test"
          onClick={[Function]}
        >
          test 2
        </a>
      </li>
    </ul>
  </nav>
  <div
    style={
      Object {
        "overflowY": "auto",
        "width": "calc(100% - 320px)",
      }
    }
  >
    <div />
  </div>
</div>
`;
